lib/engine/game/base.rb

```diff
@@ -1042,8 +1043,7 @@ def game_end_check
           end
         end
 
-        return :final_or_set, :full_or if @round.is_a?(Round::Operating) &&
-          self.class::GAME_END_CHECK[:final_or_set]&.to_i == turn
+        nil
```

Comment:
why do you need to return nil here

Reply:
The reason is that the construction that ends on line 1043 is not nil. When line 1045 was removed game_end_check breaks as any non-nil value will trigger end game. With 1045 left in it either returned something or resulted in nil.